ducktape/command_line/main.py

```diff
@@ -102,12 +43,41 @@ def extend_import_paths(paths):
     """
     for path in paths:
         dir = os.path.abspath(path if os.path.isdir(path) else os.path.dirname(path))
-        while(os.path.exists(os.path.join(dir, '__init__.py'))):
+        while os.path.exists(os.path.join(dir, '__init__.py')):
             dir = os.path.dirname(dir)
         sys.path.append(dir)
 
 
-def setup_results_directory(results_root, new_results_dir):
+def get_user_defined_globals(globals_str):
+    """Parse user-defined globals into an immutable dict using globals_str
+
+    :param globals_str Either a file, in which case, attempt to open the file and parse the contents as JSON,
+        or a JSON string representing a JSON object. The parsed JSON must represent a collection of key-value pairs,
+        i.e. a python dict.
+    :return dict containing user-defined global variables
+    """
+    empty_global = pysistence.make_dict()
```

Comment:
Don't create this if you don't need it -- just `return pysistence.make_dict()` in the `if` block